sklearn/impute/_base.py

```diff
@@ -232,7 +232,8 @@ def _validate_input(self, X, in_fit):
             # with strategy='most_frequent' or 'constant'
             # because the list is converted to Unicode numpy array
             if isinstance(X, list) and \
-               any(isinstance(elem, str) for row in X for elem in row):
+                (isinstance(self.fill_value, str)
+                 or any(isinstance(elem, str) for row in X for elem in row)):
```

Comment:
This is using `fill_value` to determine the dtype of `X` and would cast `X` to an object dtype for input such as:

```py
[[np.nan], [1], [np.nan]]
```

I am unsure if we want to do this. The current code is considering `[[np.nan], [np.nan]]` as a numerical feature and raising an error if the `fill_value` is not numerical.

If one using `None` as the missing value, then everything still works:

```py
from sklearn.impute import SimpleImputer
imputer = SimpleImputer(strategy='constant', fill_value='UNKNONW', missing_values=None)

X_2 = [[None], [None]]
imputer.fit_transform(X_2)
```